Navitia: Add Nicaragua providers

xamanu opened the pull request:
This is similar to #170 in the sense that this work came out of a local initiative to map public transport data in a region where otherwise no information and no apps exist. The data was entered into OpenStreetMap and a GTFS file generated with [osm2gtfs](https://github.com/grote/osm2gtfs). This file is then processed by Navitia that offers the routing server for the data.

The users: 
![24686747041_7b63b1bdd7_z](https://user-images.githubusercontent.com/97706/33628364-ca492512-da00-11e7-85c7-57a12fe04e05.jpg)

The team:
![24753934806_1e6b93179d_z](https://user-images.githubusercontent.com/97706/33628376-d6ea3cf2-da00-11e7-8abc-65dfe0150960.jpg)

xamanu commented:
Before considering this Pull Request, two improvements shall be done:
* The GTFS files are currently in the process of being integrated in navitia.io, which is probably a preferred sourced over the local community's navitia server.
* Tests needs to be provided.

grote commented on enabler/src/de/schildbach/pte/ManaguaProvider.java:
```diff
@@ -0,0 +1,45 @@
+ */
+public class ManaguaProvider extends AbstractNavitiaProvider {
+    private static String API_REGION = "ni-managua";
+    private static HttpUrl API_BASE = HttpUrl.parse("http://nativia.mapanica.net/").newBuilder().addPathSegment(SERVER_VERSION).build();
```
typo?

ialokim replied:
Exactly, should be changed :)

xamanu replied:
Oh, yes. Fixed.

ialokim commented on enabler/src/de/schildbach/pte/NetworkId.java:
```diff
@@ -83,4 +83,7 @@
     GHANA,
+
+    // Nicaragua
+    ESTELI, MANAGUA,
```
remove the comma :)

xamanu replied:
Which comma?

ialokim replied:
The one after MANAGUA

xamanu replied:
It should be there, i guess. All other lines end with a comma, too.

AltNico replied:
He meant the pending comma after `MANAGUA`, but IIRC this is allowed with enums in Java.

AltNico replied:
@xamanu 
> It should be there, i guess. All other lines end with a comma, too.

That's because it's a list of Enum IDs, but all older revisions of this file did not have a pending comma which you just introduced. I think schildbach would like to be consistent and have no pending commas.

xamanu replied:
The current `master` also ends with a comma.

schildbach replied:
I'm fine with the comma at the end of the list. It helps to reduce the diff of future additions.

ialokim commented on enabler/src/de/schildbach/pte/EsteliProvider.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class EsteliProvider extends AbstractNavitiaProvider {
+    private static String API_REGION = "ni-esteli";
+    private static HttpUrl API_BASE = HttpUrl.parse("http://navitia.mapanica.net/").newBuilder().addPathSegment(SERVER_VERSION).build();
```
Should be `https` to avoid redirecting.

ialokim commented on enabler/src/de/schildbach/pte/ManaguaProvider.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class ManaguaProvider extends AbstractNavitiaProvider {
+    private static String API_REGION = "ni-managua";
+    private static HttpUrl API_BASE = HttpUrl.parse("http://navitia.mapanica.net/").newBuilder().addPathSegment(SERVER_VERSION).build();
```
same here :)

ialokim commented:
We've switched now to navitia.io, where the two GTFS files for Managua and Esteli have been integrated into the same coverage, so that we now only have a NicaraguaProvider.

I've added some live tests, too.

ialokim commented:
@xamanu I think this could be renamed from WIP to merge-ready :+1:

xamanu commented:
Indeed! Now that the Nicaraguan providers [have been integrated into navitia.io](https://twitter.com/navitia/status/964209784491687936) this is ready to be reviewed by the repository owner (@schildbach) and can hopefully merged in.

I just rebased it on the most recent `master` branch, cleaned up the commit history (like removing my commit which became irrelevant) and removed the `WIP` from the title.

ialokim commented:
> We've switched now to navitia.io, where the two GTFS files for Managua and Esteli have been integrated into the same coverage, so that we now only have a NicaraguaProvider.

There are news about this one: Navitia.io [now also includes national transport](https://twitter.com/navitia/status/1006086293422575616) as [published by mapanica.net](https://github.com/mapanica/pt-data-nicaragua) including ferry services from and to Ometepe!

mayorgatellez commented:
I can confirm, as someone who lives in Managua, that the data for this city is correct and works fine with Transportr.

Thank you.

porfiriopaiz commented:
I can confirm from the ground that the data is correct and it works fine on Transportr.

roirobo commented:
I've tested the data and can confirm that it is correct and it works like a charm.

schildbach commented:
Merged. Thanks for sharing this, and thanks for the testing!

xamanu commented:
Thanks, @schildbach for merging in. Are there any additional steps necessary to include it into the Oeffi-App? Would it be also possible to include there [Managua's schematic map/plan](https://rutas.mapanica.net/mapa/)?
